src/libs/sdlcd/meson.build

```diff
@@ -0,0 +1,17 @@
+libsdlcd_sources = files([
+  'SDL_cdrom.c',
+  'win32/SDL_syscdrom.c',
+  'linux/SDL_syscdrom.c',
+  'macosx/SDL_syscdrom.c',
+  'macosx/AudioFilePlayer.c',
+  'macosx/AudioFileReaderThread.c',
+  'macosx/CDPlayer.c',
+  'macosx/SDLOSXCAGuard.c',
+  'dummy/SDL_syscdrom.c',
+])
+
```

Comment:
We can use the build system the to add the OS-specific files, with something like this:

``` python
libsdlcd_sources = ['SDL_cdrom.c']

if host_machine.system() in ['windows', 'cygwin']
  libsdlcd_sources += ['win32/SDL_syscdrom.c']

elif host_machine.system() == 'linux'
  libsdlcd_sources += ['linux/SDL_syscdrom.c']

elif host_machine.system() == 'darwin'
  libsdlcd_sources += ['macosx/SDL_syscdrom.c',
                       'macosx/AudioFilePlayer.c',
                       'macosx/AudioFileReaderThread.c',
                       'macosx/CDPlayer.c',
                       'macosx/SDLOSXCAGuard.c']

else
  libsdlcd_sources += ['dummy/SDL_syscdrom.c']
endif

# Convert list of strings to file objects
libsdlcd_sources = files(libsdlcd_sources)
```

The `host_machine` object is always available, along with a couple more as part of Meson's built-ins:
https://mesonbuild.com/Reference-manual_builtin.html

Reply:
Thanks for the suggestion. Done.